programs/manifest/src/state/market.rs

```diff
@@ -0,0 +1,1077 @@
+                matched_price.checked_quote_for_base(matched_num_base_atoms, is_bid)?;
+
+            trace!("MATCH base:{matched_num_base_atoms} price:{matched_price} quote:{quote_atoms_traded}");
+
+            // If it is a global order, just in time bring the funds over, or
+            // remove from the tree and continue on to the next order.
+            if other_order.is_global() {
+                let resting_order_trader: Pubkey =
+                    get_helper::<RBNode<ClaimedSeat>>(dynamic, other_order.get_trader_index())
+                        .get_value()
+                        .trader;
+                let global_trade_accounts_opt: &Option<GlobalTradeAccounts> = if is_bid {
+                    &global_trade_accounts_opts[0]
+                } else {
+                    &global_trade_accounts_opts[1]
+                };
+                move_global_tokens_and_modify_resting_order(
+                    global_trade_accounts_opt,
+                    &resting_order_trader,
+                    // TODO: Fix for rounding
+                    GlobalAtoms::new(if is_bid {
+                        quote_atoms_traded.as_u64()
+                    } else {
+                        matched_num_base_atoms.as_u64()
+                    }),
+                )?;
+            }
+            let other_order: &RestingOrder =
+                get_helper::<RBNode<RestingOrder>>(dynamic, current_order_index).get_value();
+
+            total_base_atoms_traded += matched_num_base_atoms;
+            total_quote_atoms_traded += quote_atoms_traded;
+
+            // These are only used when is_bid, included up here for borrow checker reasons.
+            let previous_maker_quote_atoms_allocated: QuoteAtoms =
+                matched_price.checked_quote_for_base(other_order.get_num_base_atoms(), false)?;
+            let new_maker_quote_atoms_allocated: QuoteAtoms = matched_price
+                .checked_quote_for_base(
+                    other_order
+                        .get_num_base_atoms()
+                        .checked_sub(matched_num_base_atoms)?,
+                    false,
+                )?;
+            // Use a cloned version to avoid borrow checker issues.
+            let mut cloned_other_order: RestingOrder = other_order.clone();
+
+            // Increase maker from the matched amount in the trade.
+            update_balance_by_trader_index(
+                dynamic,
+                other_trader_index,
+                !is_bid,
+                true,
+                if is_bid {
+                    quote_atoms_traded.into()
+                } else {
+                    matched_num_base_atoms.into()
+                },
+            )?;
+
+            // Possibly increase bonus atom maker gets from the rounding the
+            // quote in their favor. They will get one less than expected when
+            // cancelling because of rounding, this counters that. This ensures
```

Comment:
why would cancel do this?

Reply:
Cancel does not get paid the extra from rounding.
Will be reviewing all the rounding code again anyways after global got submitted and the redo of pricing format

Reply:
you mean it doesn't get paid the extra when a maker order gets partially filled, then gets canceled? why would a order placed, never matched and canceled get any extra?

Reply:
Because we took the extra when the order was placed